File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function maxDepth(root: TreeNode | null): number {
    // If root is null, return 0
    if (root === null) return 0;

    // Otherwise, return max between recursive left/right calls + 1
    return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
};

File: 104-maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun maxDepth(root: TreeNode?): Int {
        // If root is null, return 0
        if (root == null) return 0

        // Otherwise, return max between recursive left/right calls + 1
        return max(maxDepth(root.left), maxDepth(root.right)) + 1
    }
}